internal/pkg/mapper/scheduler/remote.go

```diff
@@ -5,24 +5,48 @@ import (
 )
 
 func (m *schedulerMapper) OnRemoteChange(changes *model.RemoteChanges) error {
-	pool := m.api.NewPool()
+	var saved []*model.ConfigState
+	var deleted []*model.ConfigState
 
 	// Activate scheduler on remote save
 	for _, objectState := range changes.Saved() {
 		if m.isSchedulerConfigFromMainBranch(objectState) {
-			m.onRemoteSave(pool, objectState.(*model.ConfigState))
+			saved = append(saved, objectState.(*model.ConfigState))
 		}
 	}
 
 	// Deactivate scheduler on remote delete
 	for _, objectState := range changes.Deleted() {
 		if m.isSchedulerConfigFromMainBranch(objectState) {
-			m.onRemoteDelete(pool, objectState.(*model.ConfigState))
+			deleted = append(deleted, objectState.(*model.ConfigState))
 		}
 	}
 
-	// Run requests
-	return pool.StartAndWait()
+	if len(saved) > 0 || len(deleted) > 0 {
+		// Get Scheduler API - only if it is needed
+		api, err := m.SchedulerApi()
+		if err != nil {
+			return err
+		}
```

Comment:
SchedulerApi sa teraz ziskava `lazy`. Jedna sa usetri jeden request na `services`, ak `SchedulerApi` nepotrebujeme.
Ale hlavne nam to umoznuje robit rozne operacie nad `template`, napr. validaciu, bez toho aby sme sa museli pripojit k API a mat Storage API token -> sice to este nie je uplne mozne, lebo nemame `lazy` nacitavanie `Storage API`, ale je to krok tymto smerom.